### DISCUSSION TITLE
"init_data_sys" and "sys_configs" in dpgen (lammps-cp2k) example param.json file

### DISCUSSION THREAD

**Discussion by longkunxuluke** [deepmodeling/dpgen#993] [Q&A]
Hi, I am trying to use dpgen working with LAMMPS and CP2K (without VASP and others). I follow [this example](https://github.com/deepmodeling/dpgen/blob/master/examples/run/dp1.x-lammps-cp2k/methane/param-ch4.json) but not sure how to properly set the two parameters in the title. 

Instead of using ```dpgen init```, my initial data was produced using dpdata to process CP2K AIMD output following [this tutorial](https://github.com/deepmodeling/deepmd-kit/blob/master/doc/data/dpdata.md). Should I put the directory containing the generated ```deepmd/npy``` files in "init_data_sys"? Should I randomly pick 4 configurations from AIMD trajectory and convert them to POSCAR files and put their directory in "sys_configs"? 

I am using a local workstation, without using VASP and ```dpgen init```, I tried looking for example param.json and machine.json specifically for this scenario but have not found. Any help would be appreciated.

**Comment by HuangJiameng** [comment]
https://docs.deepmodeling.com/projects/dpgen/en/devel/user-guide/troubleshooting.html Hope it is helpful.

**Reply by longkunxuluke** [reply]
Thanks for this. I checked the troubleshooting list and I think there's not in my case. I met an error with 

```
dpgen/lib/python3.8/site-packages/dpgen/remote/decide_machine.py", line 39, in convert_mdata
    for key, item in mdata[task_type][0].items():
KeyError: 0

```
I think this may be related with ```machine.json```

**Reply by njzjz** [reply]
Could you provide `machine.json`?

**Reply by longkunxuluke** [reply]
Please check the attached, thanks
[json.zip](https://github.com/deepmodeling/dpgen/files/9750446/json.zip)

**Reply by njzjz** [reply]
The file is correct, but you do not use the latest version.

**Reply by longkunxuluke** [reply]
Sorry did you mean the latest version of dpgen, deepmd-kit, or the formats in my json files?

**Reply by njzjz** [reply]
`decide_machine.py` is different from that on the GitHub.
https://github.com/deepmodeling/dpgen/blob/8dea29ef125f66be9641afe5ac4970433a9c9ce1/dpgen/remote/decide_machine.py#L39

**Comment by njzjz** [comment]
[`sys_configs`](https://docs.deepmodeling.com/projects/dpgen/en/devel/run/param.html#argument:run_jdata/sys_configs) is the starting structures for MD simulations. You can set another [`sys_format`](https://docs.deepmodeling.com/projects/dpgen/en/devel/run/param.html#argument:run_jdata/sys_format) if you don't want POSCAR.

Supported format can be found in the [dpdata documentation](https://docs.deepmodeling.com/projects/dpdata/en/master/formats.html).

**Reply by chenyuping6** [reply]
I also want to ask, how to select structures from AIMD trajectories and convert them into POSCAR files to place in "sys_configs"?
